Sources/SpelledPitch/PitchSpeller/Wetherfield/WeightedGraphSchemeProtocol.swift

```diff
@@ -38,51 +38,35 @@ extension WeightedGraphSchemeProtocol {
 extension WeightedGraphSchemeProtocol where Self: UndirectedGraphSchemeProtocol, Weight: Numeric {
     
     static func * (lhs: Self, rhs: Self) -> Self {
-        return Self { edge in
-            guard let lweight = lhs.weight(edge), let rweight = rhs.weight(edge) else { return nil }
-            return lweight * rweight
-        }
+        return Self { edge in lhs.weight(edge) ?? rhs.weight(edge) }
     }
     
     static func + (lhs: Self, rhs: Self) -> Self {
-        return Self { edge in
-            if let lweight = lhs.weight(edge) {
-                if let rweight = rhs.weight(edge) { return lweight + rweight }
-                return lweight
-            } else {
-                guard let rweight = rhs.weight(edge) else { return nil }
-                return rweight
-            }
+        return Self.init { edge in
+            return lhs.weight(edge).flatMap { lhs in rhs.weight(edge).map { rhs in lhs + rhs } }
```

Comment:
If `rhs.weight(edge)` is `nil`, don't we get `nil` instead of `lhs` ?

Reply:
Yep. Wrong here too!

Reply:
Used this one for multiplication in the other branch, where I think this pattern works!